src/test/scala/pt/tecnico/dsi/neutron/CrudSpec.scala

```diff
@@ -1,65 +1,69 @@
 package pt.tecnico.dsi.neutron
 
-import cats.effect.IO
+import cats.effect.{IO, Resource}
 import cats.implicits._
-import cats.syntax.flatMap._
 import io.circe.{Decoder, Encoder}
-import pt.tecnico.dsi.neutron.models.Model
+import org.scalatest.Assertion
+import pt.tecnico.dsi.neutron.models.{Model, Network}
 import pt.tecnico.dsi.neutron.services.{BulkCreate, CrudService}
 import pt.tecnico.dsi.openstack.common.models.WithId
 
-abstract class CrudSpec[T <: Model](val name: String, val service: NeutronClient[IO] => CrudService[IO, T])
+abstract class CrudSpec[T <: Model](val name: String)
   (implicit val encoder: Encoder[T#Create], val decoder: Decoder[WithId[T#Read]])
   extends Utils {
 
+  val service: CrudService[IO, T]
+
   val displayName: String = name.capitalize
-  val createStub: IO[T#Create]
+  val updateStub: IO[T#Update]
+  def updateComparator(read: T#Read, update: T#Update): Assertion
 
-  val withSubCreated: IO[(WithId[T#Read], CrudService[IO, T])] =
-    for {
-      neutron <- client
-      stub <- createStub
-      crudService = service(neutron)
-      createdStub <- crudService.create(stub)
-    } yield (createdStub, crudService)
+  val withStubCreated: Resource[IO, WithId[T#Read]]
 
+  val withNetworkCreated: Resource[IO, WithId[Network.Read]] = {
+    val created = client.networks.create { Network.Create() }
+    Resource.make(created)(x => client.networks.delete(x.id))
+  }
 
   s"$displayName service" should {
 
```

Comment:
The tests are not testing idempotency. Here is an example https://github.com/ist-dsi/scala-designate-client/blob/master/src/test/scala/pt/tecnico/dsi/openstack/designate/RecordsetsSpec.scala